runtime/src/accounts.rs

```diff
@@ -1033,28 +1133,169 @@ mod tests {
         );
     }
 
+    #[test]
```

Comment:
Could we add a multi-threaded test here with multiple iterations of: 

For the same account:
1) Some threads continually tries to grab credit-debit locks, increment some atomic `counter` when they succeed.
2) Some threads continually try to grab credit-only locks, hold them for a while, release them. They should observe that the `counter` value never changes while they hold the lock (implies that credit-debit was excluded)